### DISCUSSION TITLE
Protobuf Integration

### DISCUSSION THREAD

**Discussion by eadlam** [cue-lang/cue#1113] [General]
cuelang.org lists some planned features for the [protobuf integration](https://cuelang.org/docs/integrations/protobuf/):

- generating Protobuf definitions from CUE definitions
- encoding text, binary and JSON Protobuf messages from CUE
- decoding text, binary and JSON Protobuf messages to CUE.

are these still on the roadmap?

**Comment by myitcv** [comment]
Apologies for the delay in replying, @eadlam.

> * generating Protobuf definitions from CUE definitions

This falls under the heading of a conversion that has not yet been implemented, but is very much on the horizon. We will be creating placeholder issues for all "missing" or planned conversions; I will cc you on the CUE -> protobuf one.

> * encoding text, binary and JSON Protobuf messages from CUE
> * decoding text, binary and JSON Protobuf messages to CUE.

These should be covered by the `v0.4.0` release. The [`v0.4.0` release notes](https://github.com/cue-lang/cue/releases/tag/v0.4.0) link to more detail on this point in the [`v0.4.0-alpha.1` release notes](https://github.com/cuelang/cue/releases/tag/v0.4.0-alpha.1).

**Reply by jabinb** [reply]
I too am interested on doing CUE -> protobuf, if you could cc me as well that would be great.

**Reply by verdverm** [reply]
maybe it would be easiest for @myitcv to make comments here and have people subscribe to the discussion? (Button on the RHS here under "notifications")

You may be subscribed by virtue of commenting on this discussion already.

**Reply by myitcv** [reply]
Thanks @verdverm - I will indeed reply back here with the issue when I create it to ensure everything is linked up.
